### ISSUE TITLE
Wrong credentials when adding integration

### ISSUE THREAD

**Issue by fredrikker** [radical-squared/aquatemp#24]
**Was not able to add the integration due to bad credentials**
I added using HACS, tried to add "Aqua Temp" integration, got the popup with username and password, entered email and password and got an error stating wrong credentials. 

Checked the log files and found the following lines:
_2023-05-27 23:04:22.982 WARNING (MainThread) [custom_components.aqua_temp.managers.aqua_temp_api] Failed to initialize session, Error: Cannot serialize non-str key None, Line: 77
2023-05-27 23:04:22.988 WARNING (MainThread) [custom_components.aqua_temp.config_flow] Failed to create integration, Error: invalid_credentials_

So changed line 52 in custom_components.aqua_temp.managers.aqua_temp_api:
self._headers = HEADERS
to:
self._headers = {str(key): value for key, value in HEADERS.items()}

Which resulted in a successful message where the integration was added. I'm able to control the heat pump and get correct temperature readings.

Thanks to ChatGPT since I've never programmed in python. Thought I should share this if someone else experience the same issue.

**To Reproduce**

1. Go to Settings - Devices & Services
2. Click on '+ ADD INTEGRATION'
3. Find Aqua Temp
4. Enter email & password
5. Check error

**Expected behavior**
I expected that the credentials would work, double checked my credentials used in the app.

**Aqua Temp Custom Component**
- Version: v.3.0.1

**Desktop**
- OS: Win 10 Pro
- Browser: Chrome 113.0.5672.127 (Official Build) (64-bit)

**Comment by sjtrny** [comment]
That's really strange and I suspect something else was the problem because your new code produces the same result. See below for demo.

```
Python 3.11.3 (main, Apr  7 2023, 21:05:46) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> HEADERS = {"Content-Type": "application/json; charset=utf-8"}
>>> NEW_HEADERS = {str(key): value for key, value in HEADERS.items()}
>>> HEADERS == NEW_HEADERS
True
```

**Comment by elad-bar** [comment]
Expirenced the same issue, fixed it for v3.0.2,
It seems that once the token is failed to be generated, it left out in memory which prevents creating new one,
Soluyion is much simpler and requires to delete both variable and headers object for next login attempt.

Upcoming version is in testing, will publish it later today / tomorrow

Thanks

**Comment by fredrikker** [comment]
Awesome! Makes sense why it failed for just some situations.
